src/Tasks/Hash.cs

```diff
@@ -37,61 +35,120 @@ public class Hash : TaskExtension
+                    }
 
-                        stringBuilder.Clear();
+                    sha1.TransformFinalBlock(sha1Buffer, 0, bufferLength);
 
-                        foreach (var b in hash)
+                    using (var stringBuilder = new ReuseableStringBuilder(sha1.HashSize))
+                    {
+                        foreach (var b in sha1.Hash)
                         {
                             stringBuilder.Append(b.ToString("x2"));
                         }
-
                         HashResult = stringBuilder.ToString();
                     }
                 }
             }
-
             return true;
         }
 
-        private int ComputeStringSize(ITaskItem[] itemsToHash)
+        /// <summary>
+        /// Add bytes to the sha1 buffer. Once the limit size is reached, sha1.TransformBlock is called and the buffer is flushed.
+        /// </summary>
+        /// <param name="sha1">Hashing algorithm sha1.</param>
+        /// <param name="sha1Buffer">The sha1 buffer which stores bytes of the strings. Bytes should be added to this buffer.</param>
+        /// <param name="sha1BufferLength">Number of used bytes of the sha1 buffer.</param>
+        /// <param name="sha1BufferSize">The size of sha1 buffer.</param>
+        /// <param name="bytesBuffer">Bytes buffer which contains bytes to be written to sha1 buffer.</param>
+        /// <param name="bytesNumber">Amount of bytes that are to be added to sha1 buffer.</param>
+        /// <returns></returns>
+        private bool AddBytesToSha1Buffer(SHA1 sha1, byte[] sha1Buffer, ref int sha1BufferLength, int sha1BufferSize, byte[] bytesBuffer, int bytesNumber)
         {
-            if (itemsToHash.Length == 0)
+            int bytesProcessedNumber = 0;
+            while (sha1BufferLength + bytesNumber >= sha1BufferSize)
             {
-                return 0;
+                int sha1BufferFreeSpace = sha1BufferSize - sha1BufferLength;
+
+                if (sha1BufferLength == 0)
+                {
+                    // If sha1 buffer is empty and bytes number is big enough there is no need to copy bytes to sha1 buffer.
+                    // Pass the bytes to TransformBlock right away.
+                    sha1.TransformBlock(bytesBuffer, 0, sha1BufferSize, null, 0);
+                }
+                else
+                {
+                    Array.Copy(bytesBuffer, bytesProcessedNumber, sha1Buffer, sha1BufferLength, sha1BufferFreeSpace);
+                    sha1.TransformBlock(sha1Buffer, 0, sha1BufferSize, null, 0);
+                    sha1BufferLength = 0;
+                }
+
+                bytesProcessedNumber += sha1BufferFreeSpace;
+                bytesNumber -= sha1BufferFreeSpace;
             }
 
-            var totalItemSize = 0;
+            Array.Copy(bytesBuffer, bytesProcessedNumber, sha1Buffer, sha1BufferLength, bytesNumber);
+            sha1BufferLength += bytesNumber;
 
+            return true;
```

Comment:
Do you ever use this?

Reply:
Right, forgot to remove the return value. I will make it void.